lib/CBonusTypeHandler.cpp

```diff
@@ -99,60 +99,66 @@ std::string CBonusTypeHandler::bonusToGraphics(const std::shared_ptr<Bonus> & bo
 		fileName = sp->getIconImmune();
 		break;
 	}
-	case BonusType::FIRE_IMMUNITY:
+	case BonusType::SPELL_DAMAGE_REDUCTION: //Spell damage reduction for all schools

```

Comment:
And this is why we need way to configure bonus icons at least based on subtype.
Hardcoding everything is not the best idea.

Reply:
Agree. Selector-based config will be after part 2 (psychic and magic elementals).